drivers/gpio/gpio_ite_it8xxx2.c

```diff
@@ -0,0 +1,465 @@
+#define GPIO_GPDRF		(DT_REG_ADDR_BY_IDX(DT_NODELABEL(gpiof), 0))
+#define GPIO_GPDRM		(DT_REG_ADDR_BY_IDX(DT_NODELABEL(gpiom), 0))
+#define GPCR_OFFSET		0x10
+#define GPIO_DIR_INPUT		0x80
+#define GPIO_DIR_OUTPUT	0x40
+
+struct gpio_ite_wui {
+	volatile uint8_t *reg_addr;
+	volatile uint8_t *clear_addr;
+	volatile uint8_t *bothedge_addr;
+	uint32_t pin;
+	uint32_t irq;
+};
+
+static const struct gpio_ite_wui GPIO_GPDRB_wui[NUM_IO_MAX] = {
+	{ &WUEMR10, &WUESR10, &WUBEMR10, BIT(5), 106 },           /* gpb0, */
+	{ &WUEMR10, &WUESR10, &WUBEMR10, BIT(6), 107 },           /* gpb1, */
+	{ &WUEMR8, &WUESR8, &WUBEMR8, BIT(4), 92 },               /* gpb2, */
+	{ &WUEMR10, &WUESR10, &WUBEMR10, BIT(7), 108 },           /* gpb3, */
+	{ &WUEMR9, &WUESR9, &WUBEMR9, BIT(6), 99 },               /* gpb4, */
+	{ &WUEMR11, &WUESR11, &WUBEMR11, BIT(0), 109 },           /* gpb5, */
+	{ &WUEMR11, &WUESR11, &WUBEMR11, BIT(1), 110 },           /* gpb6, */
+	{ &WUEMR11, &WUESR11, &WUBEMR11, BIT(2), 111 },           /* gpb7, */
+
+};
+
+static const struct gpio_ite_wui GPIO_GPDRF_wui[NUM_IO_MAX] = {
+	{ &WUEMR10, &WUESR10, &WUBEMR10, BIT(0), 101 },           /* gpf0, */
+	{ &WUEMR10, &WUESR10, &WUBEMR10, BIT(1), 102 },           /* gpf1, */
+	{ &WUEMR10, &WUESR10, &WUBEMR10, BIT(2), 103 },           /* gpf2, */
+	{ &WUEMR10, &WUESR10, &WUBEMR10, BIT(3), 104 },           /* gpf3, */
+	{ &WUEMR6, &WUESR6, &WUBEMR6, BIT(4), 52 },               /* gpf4, */
+	{ &WUEMR6, &WUESR6, &WUBEMR6, BIT(5), 53 },               /* gpf5, */
+	{ &WUEMR6, &WUESR6, &WUBEMR6, BIT(6), 54 },               /* gpf6, */
+	{ &WUEMR6, &WUESR6, &WUBEMR6, BIT(7), 55 },               /* gpf7, */
+};
+
+/* struct gpio_ite_reg_table is record different gpio port's register set
+ * base_addr: base address of gpio set
+ * wui[]: wui register group
+ */
+struct gpio_ite_reg_table {
+	uint32_t base_addr;
+	const struct gpio_ite_wui *wui;
+};
+
+struct gpio_ite_reg_table gpiox_reg[CURR_SUPP_GPIO_SET] = {
+	{GPIO_GPDRB, GPIO_GPDRB_wui},	/* GPIO GROPU B */
+	{GPIO_GPDRF, GPIO_GPDRF_wui},	/* GPIO GROPU F */
+};
+
+/* edge/level trigger register */
+static volatile uint8_t *const reg_ielmr[] = {
+	&IELMR0, &IELMR1, &IELMR2, &IELMR3,
+	&IELMR4, &IELMR5, &IELMR6, &IELMR7,
+	&IELMR8, &IELMR9, &IELMR10, &IELMR11,
+	&IELMR12, &IELMR13, &IELMR14, &IELMR15,
+	&IELMR16, &IELMR17, &IELMR18, &IELMR19,
+	&IELMR20
+};
+
+/* high/low trigger register */
+static volatile uint8_t *const reg_ipolr[] = {
+	&IPOLR0, &IPOLR1, &IPOLR2, &IPOLR3,
+	&IPOLR4, &IPOLR5, &IPOLR6, &IPOLR7,
+	&IPOLR8, &IPOLR9, &IPOLR10, &IPOLR11,
+	&IPOLR12, &IPOLR13, &IPOLR14, &IPOLR15,
+	&IPOLR16, &IPOLR17, &IPOLR18, &IPOLR19,
+	&IPOLR20
+};
+
+/*
+ * Strcture gpio_ite_cfg is about the setting of gpio
+ * this config will be used at initial time
+ */
+struct gpio_ite_cfg {
+	gpio_port_pins_t port_pin_mask;
+	uint32_t reg_addr;	/* gpio register base address */
+	char *label;		/* label is set by device tree */
```

Comment:
There is no need to store label in the config structure. Should be passed directly to `DEVICE_AND_API_INIT` macro.